src/Functions/isValidASCII.cpp

```diff
@@ -0,0 +1,144 @@
+    }
+
+    size_t getNumberOfArguments() const override
+    {
+        return 1;
+    }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & /*arguments*/) const override
+    {
+        return std::make_shared<DataTypeUInt8>();
+    }
+
+    bool useDefaultImplementationForConstants() const override
+    {
+        return true;
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        const auto & column = arguments[0];
+
+        if (const auto * col_str = checkAndGetColumn<ColumnString>(column.column.get()))
+        {
+            return executeString(col_str, input_rows_count);
+        }
+        else if (const auto * col_fixed_str = checkAndGetColumn<ColumnFixedString>(column.column.get()))
+        {
+            return executeFixedString(col_fixed_str, input_rows_count);
+        }
+        else
+        {
+            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "Cannot apply function {} to column {}, expected String or FixedString",
+                getName(), column.column->getName());
+        }
+    }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override
+    {
+        return true;
+    }
+
+private:
+    ColumnPtr executeString(const ColumnString * col_str, size_t input_rows_count) const
+    {
+        auto result = ColumnUInt8::create(input_rows_count);
+        auto & result_data = result->getData();
+
+        const auto & chars = col_str->getChars();
+        const auto & offsets = col_str->getOffsets();
+
+        size_t prev_offset = 0;
+        for (size_t i = 0; i < input_rows_count; ++i)
+        {
+            size_t current_offset = offsets[i];
+            size_t string_size = current_offset - prev_offset;
+            result_data[i] = isValidASCII(chars.data() + prev_offset, string_size);
+            prev_offset = current_offset;
+        }
+
+        return result;
+    }
+
+    ColumnPtr executeFixedString(const ColumnFixedString * col_fixed_str, size_t input_rows_count) const
+    {
+        auto result = ColumnUInt8::create(input_rows_count);
+        auto & result_data = result->getData();
+
+        const auto & chars = col_fixed_str->getChars();
+        size_t string_size = col_fixed_str->getN();
+
+        for (size_t i = 0; i < input_rows_count; ++i)
+        {
+            result_data[i] = isValidASCII(chars.data() + i * string_size, string_size);
+        }
+
+        return result;
+    }
+};
+
+}
+
+REGISTER_FUNCTION(IsValidASCII)
+{
+    factory.registerFunction<DB::FunctionIsValidASCII>(DB::FunctionDocumentation{
+        .description = R"(Returns 1 if the input String or FixedString contains only ASCII bytes (0x00–0x7F), otherwise 0.)",
+        .examples = {{"isValidASCII", "SELECT isValidASCII('hello') AS is_ascii, isValidASCII('你好') AS is_not_ascii", ""}},
+        .introduced_in = {25, 8},
```

Comment:
Should be 25.9 now. 25.8 was created before merging this. Can you update it please?

Reply:
Thanks, updating there https://github.com/ClickHouse/ClickHouse/pull/86120/files